src/WSRequestHandler_General.cpp

```diff
@@ -231,6 +231,45 @@ HandlerResponse WSRequestHandler::HandleGetStats(WSRequestHandler* req) {
 	return req->SendOKResponse(response);
 }
 
+/**
+ * Broadcast some data to all connected WebSocket clients
+ *
+ * @param {String} `realm` Some Identifier to be choosen by the client
+ * @param {Object} `data` User-defined data
+ *
+ * @api general
+ * @name BroadcastWebSocketMessage
+ * @category general
+ * @since 4.7.0
+ */
+HandlerResponse WSRequestHandler::HandleBroadcastWebSocketMessage(WSRequestHandler* req) {
```

Comment:
Can you also please add at the top of this file the specification for the associated event, please?

Reply:
personal nitpick: I believe `BroadcastCustomMessage` for the request type and `OnCustomMessage` for the event would be more fitting names

Reply:
@Palakis I changed it to BroadcastCustomMessage and CustomMessage (the other events also don't have the "On"-Prefix, only in the name of the function). I don't understand the part about "associated event", I checked the source in order to understand what you might mean but did not find out. Please advise.

Reply:
> I don't understand the part about "associated event", I checked the source in order to understand what you might mean but did not find out. Please advise.

The `CustomMessage` event must be documented like the events in `WSEvents.cpp`. Events spec example: https://github.com/Palakis/obs-websocket/blob/4.x-current/src/WSEvents.cpp#L495-L504

You can either leave the `CustomMessage` event here and document it at the top of this file, or move it into a `WSEvents::OnCustomMessage` method in `WSEvents.cpp` (with `BroadcastCustomMessage` calling it from here) and document it there.

Reply:
The spec for `CustomMessage` would look something like this:

```
/**
 * Custom message from another client.
 *
 * @return {String} `realm` User-defined message identifier
 * @return {Object} `data` User-defined data
 *
 * @api events
 * @name CustomMessage
 * @category general
 * @since 4.7.0
 */
```

Reply:
I have splitted it.

Reply:
I used the name OnBroadcastCustomMessage still because as suggested before it might be nice to also have custom messages which are not broadcasted but directed at a target.

Reply:
Agreed! :+1: